File: CS323/CS323-DZ11/CS323-DZ11/Osoba.cpp
#include "Osoba.h"

Osoba::Osoba() {

}

Osoba::Osoba(string ime, string prezime, string adresa, string telefon) {
	this->ime = ime;
	this->prezime = prezime;
	this->adresa = adresa;
	this->telefon = telefon;
}

Osoba::~Osoba() {
	cout << "Osoba je obrisana!" << endl;
}

void Osoba::setIme(string ime) {
	this->ime = ime;
}

void Osoba::setPrezime(string prezime) {
	this->prezime = prezime;
}

void Osoba::setAdresa(string adresa) {
	this->adresa = adresa;
}

void Osoba::setTelefon(string telefon) {
	this->telefon = telefon;
}


string Osoba::getIme() const {
	return ime;
}
string Osoba::getPrezime() const {
	return prezime;
}
string Osoba::getAdresa() const {
	return adresa;
}

string Osoba::getTelefon() const {
	return telefon;
}

string Osoba::toString() {
	string str = ime;
	str += " " + prezime;
	str += " " + adresa;
	str += " " + telefon;
	return str;
}

File: CS324-V01-January/zadatak2/osoba.py
from abc import ABC, abstractmethod


# Abstraktna klasa
class Osoba(ABC):
    def __init__(self, ime, prezime):
        self.ime = ime
        self.prezime = prezime

    @abstractmethod
    def set_ime(self, ime):
        self.ime = ime

    @abstractmethod
    def set_prezime(self, prezime):
        self.prezime = prezime

    @abstractmethod
    def get_ime(self):
        return self.ime

    @abstractmethod
    def get_prezime(self):
        return self.prezime
